### ISSUE TITLE
There is an extra number before <PRI> 

### ISSUE THREAD

**Issue by dbrosluc** [IonxSolutions/serilog-sinks-syslog#43]
Hello,

I am trying to use serilog.sinks.syslog to send RFC5424 formatted message via TCP. I always get an extra number comes before <PRI>. For example:
335 <134>1 2021-12-13T09:36:49.966549-08:00 MM123 AppLogging 28192 ...

My question is: what is 335 number represent? My logstash server does not recognize it. Is there a way to configure not to have this number in our message?

Thank you so much for your help.

**Comment by Kevin-Andrew** [comment]
That is the framing part of the transmission protocol, in this case, **Octet Counting**. RFC 5424 is just for the format of the message, for example the `<priority>` and the date/time format. It is not for the transmission of the data over the wire, in particular, a TCP connection. For that, you have RFC 6587, in particular https://datatracker.ietf.org/doc/html/rfc6587#section-3.4.1.

RFC 6587 also mentions other, older, problematic, message framing techniques that have or are still used, namely `LF` or `NUL` (and there are even some that offer/use `CR` or `CRLF`). You can imagine the problems that those may cause with a streaming protocol like TCP. If a message were to include a `LF` character, your server would incorrectly treat it as the end of a message.

For example, in the Windows program "Syslog Watcher", it has the available options:
![image](https://user-images.githubusercontent.com/22459191/145948992-b02f67c4-1fdf-4b45-b064-cc496dcabcea.png)
Unfortunately, **Octet Counting** is not one of them.

Because of that, and other things, I ended up writing my own Syslog Server Listener, and you can see the options that I provide:
![image](https://user-images.githubusercontent.com/22459191/145949309-7a06f059-2a88-4ee0-8d77-5ebf0953086f.png)

Are you by chance trying to write you own Syslog server? Or did you find the Octet Counting bytes by debugging?

Having said all of that, there is an option in this Serilog.Sinks.Syslog component to control the framing of a message sent over TCP. See the [FramingType](https://github.com/IonxSolutions/serilog-sinks-syslog/blob/v2.0.6/src/Serilog.Sinks.Syslog/Sinks/Settings/FramingType.cs) and [MessageFramer](https://github.com/IonxSolutions/serilog-sinks-syslog/blob/v2.0.6/src/Serilog.Sinks.Syslog/Sinks/MessageFramer.cs).

Depending on how you go about creating your `LoggerConfiguration`, there is, for example, a parameter of one of the extension methods to control/set the framing type:
https://github.com/IonxSolutions/serilog-sinks-syslog/blob/v2.0.6/src/Serilog.Sinks.Syslog/SyslogLoggerConfigurationExtensions.cs#L130

You may also have the option to change the framing type in the Syslog server application.

To recap, based on the information you've provided, this is not a bug. It is just the default, recommended behavior/framing type.

Hope that helps. If there isn't anything else, we will close this issue.

**Comment by dbrosluc** [comment]
Hello Kevin-Andrew,

Thank you so much for your reply and very helpful information. I did not understand what was that initial number but with your explanation, I am able to use Serilog FramingType to change it to CR and it works. Please close the issue.
